backend/lucene/src/main/java/org/hibernate/search/backend/lucene/lowlevel/directory/impl/LocalFileSystemDirectoryProvider.java

```diff
@@ -58,7 +58,7 @@ public void initialize(DirectoryProviderInitializationContext context) {
 		this.directoryRoot = ROOT.get( propertySource ).toAbsolutePath();
 		FileSystemAccessStrategyName accessStrategyName = FILESYSTEM_ACCESS_STRATEGY.get( propertySource );
 		this.accessStrategy = FileSystemAccessStrategy.get( accessStrategyName );
-		this.lockFactory = context.createConfiguredLockFactory().orElseGet( FSLockFactory::getDefault );
```

Comment:
Why do you call `get` here? You'll end up having the exact same problem as with the local heap directory, if the user selects the single-instance lock factory explicitly through configuration.